application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java

```diff
@@ -169,34 +170,50 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
 
 
         String applicationId;
+        String applicationName;
 
         if (applicationOption != null) {
-            applicationId = VC_APPLICATION_TO_ID.get(applicationOption.getAsString());
+            applicationName = applicationOption.getAsString();
+            applicationId = VC_APPLICATION_TO_ID.get(applicationName);
         } else {
             applicationId = idOption.getAsString();
+
+            applicationName =
+                    getKeyByValue(VC_APPLICATION_TO_ID, applicationId).orElse("an activity");
+        }
+
+        handleSubcommand(event, voiceChannel, applicationId, maxUses, maxAge, applicationName);
+    }
+
+    private static <K, V> Optional<K> getKeyByValue(@NotNull Map<K, V> map, @NotNull V value) {
```

Comment:
NotNull on return type as well :)

Reply:
Fixed in 52d86d77b2f17896138b8bc16f30200de343664e